### ISSUE TITLE
Cannot Find -lqtadvanceddocking (Linking the ADS as a Shared/Dynamic Library)

### ISSUE THREAD

**Issue by LiamC12345** [githubuser0xFFFF/Qt-Advanced-Docking-System#391]
I did everything required to link the ADS as a shared library and am getting the following error when building my Project:
cannot find -lqtadvanceddocking

Has anyone here had this issue and found a solution?

**Comment by githubuser0xFFFF** [comment]
Please follow the instructions in the Qt documentation to learn how to use shared libraries. This is not an ADS issue.

**Comment by LiamC12345** [comment]
But this same common issue occurs straight out of the box upon unzipping the Qt-Advanced-Docking-System-master.zip
I've tried this on two different computers (Mac and PC had exactly the same issue).

Try it for yourself - it takes 2 seconds
If you unzip a new instance from Qt-Advanced-Docking-System-master.zip from scratch and then try to build either the
demo or any of one the examples you will get the same issue. 

Do this WITHOUT building the ads.pro at all, then you'll see what I mean.
For instance, only open and build simple.pro in the examples and try to run it.
Guarantee you'll get the -lqtadvanceddocking error

**Comment by LiamC12345** [comment]
I'm about 90% sure this is an ADS issue because it occurs straight out of the box without tampering anything
on two different computers so far.

I'm telling you, I really did learn how to use shared libraries - I can spell out the entire step by step process of it for you if you
don't believe me.

Additional ideas I have regarding this issue if this helps:
I've tried both Debug and Release Builds Settings and I don't think it's a Mingw/MSVC thing,
surely this should just work with Mingw only right?
Could it possibly be a Qt 6. Compatibility thing? If so it means the ADS isn't fully compatible with it.

**Comment by githubuser0xFFFF** [comment]
The library and all examples and demos are build automatically on the Github CI system. The green badges on the main page indicate that everything works fine so you are likely doing something wrong.

I also checked out the master and tested the build locally for you - it works.

**Comment by LiamC12345** [comment]
Thankyou for testing that,
It's just that I worked on it for hours and hours, staying up till 3am last night doing everything I could to try and debug it.
I googled, youtube'd and read through every Qt Documentation I could find on the matter.
There comes a point where there's only so much research one can do and still get nowhere. 

But I guess I'll just keep trying different things and hope for the best
Thanks again

**Comment by LiamC12345** [comment]
Holy Shit! You won't believe it! I got it working!
I redownloaded the .zip file and noticed a slight size difference between the master zip I was using vs the new download

I think it's because I downloaded the Qt-Advanced-Docking-System-master.zip ages ago when there was a bug and it as been updated since

HALLELUJAH!!!!!!

Thankyou for putting up with all my nightmares!!!
God Bless!!!

I highly doubt I'll need to annoy you for help here now - all is right with the world again XD
